data-products/src/main/scala/org/sunbird/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -0,0 +1,280 @@
+package org.sunbird.analytics.exhaust
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Encoders, SQLContext, SparkSession}
+import org.apache.spark.sql.functions.{col, lit, when}
+import org.ekstep.analytics.framework.Level.INFO
+import org.ekstep.analytics.framework.conf.AppConf
+import org.ekstep.analytics.framework.util.{CommonUtil, JSONUtils, JobLogger, MergeUtil, RestUtil}
+import org.ekstep.analytics.framework.{DruidQueryModel, FrameworkContext, JobConfig, JobContext, StorageConfig, _}
+import org.ekstep.analytics.model.{OutputConfig, QueryDateRange, ReportMergeConfig}
+
+import java.util.concurrent.atomic.AtomicInteger
+import scala.collection.immutable.{List, Map}
+import scala.collection.mutable
+import scala.collection.mutable._
+import org.ekstep.analytics.framework.exception.DruidConfigException
+import org.ekstep.analytics.framework.fetcher.DruidDataFetcher
+import org.ekstep.analytics.framework.util.DatasetUtil.extensions
+import org.ekstep.analytics.model.DruidQueryProcessingModel.{getDateRange, getReportDF}
+import org.apache.hadoop.conf.Configuration
+
+import java.util.concurrent.CompletableFuture
+import java.util.function.Supplier
+
+
+case class RequestBody(`type`: String,`params`: Map[String,AnyRef])
+case class DruidQuery(queryType: String, dataSource: String, intervals: String, granularity: Option[String], aggregations: Option[List[Aggregation]],
+                      dimensions: Option[List[DruidDimension]], filters: Option[List[mutable.Map[String,AnyRef]]], having: Option[DruidHavingFilter],
+                      postAggregation: Option[List[PostAggregation]], columns: Option[List[String]], sqlDimensions: Option[List[DruidSQLDimension]],
+                      threshold: Option[Long], metric: Option[String], descending: Option[String], intervalSlider: Int)
+case class Metrics(metric : String, label : String, druidQuery : DruidQuery)
+case class ReportConfig(id: String, queryType: String, dateRange: QueryDateRange, metrics: List[Metrics], labels: LinkedHashMap[String, String],
```

Comment:
ReportConfig, Metrics and DruidQuery case classes can be directly used by importing from Framework/CoreDataProducts and no need to define once again here.

Reply:
ReportConfig changes is done. other part we are working.

Reply:
Metrics and DruidQuery case classes is also done